src/design-pages/passkey_kerberos.rst

```diff
@@ -0,0 +1,230 @@
+  communicate with IPA server side (ipa-otpd daemon listens over UNIX domain
+  socket used by the KDC). The RADIUS message Access-Request will be sent to
+  ipa-otpd.
+
+* ipa-otpd will retrieve the user information and generate the assertion
+  request data (more information on the format in
+  :ref:`kerberos-kdc-ipa-otpd`), which will be used to fill the Reply-Message
+  that is returned to the Kerberos KDC.
+
+* The passkey assertion request data produced by the KDC side of the
+  pre-authentication method implementation is returned to libkrb5 in an error
+  response.
+
+* This information will be passed between the various components until it
+  reaches the PAM responder. It will request PAM to get the PIN from the user
+  and it will execute the passkey_child helper process with the assertion
+  request data and the PIN as arguments.
+
+* The passkey_child will request the assertion with the assertion request data
+  and the PIN.
+
+* The passkey will return the assertion data, which will also be returned by
+  the passkey_child.
+
+* The PAM responder will fill the pre-authentication with the assertion data.
+  If the credential is discoverable, then the ``userId`` will be included in
+  the assertion data. The PAM responder will check if it matches with the one
+  stored in the LDAP attribute, and it will fail if they don't match. 
+
+* libkrb5 will send another AS-REQ but this time with the assertion in the
+  pre-authentication.
+
+* The Kerberos KDC will generate the second Access-Request message, which will
+  contain the assertion data.
+
+* ipa-otpd will ask the passkey_child that is located in the FreeIPA server to
+  validate the assertion data. It will reply with Access-Accept/Reject
+  depending on the authentication result.
+
+* If the validation is successful the Kerberos KDC will issue a ticket and wrap
+  it in AS-REP.
+
+* krb5_child will store the ticket and return the authentication success to the
+  IPA provider. This information will be propagated by the PAM responder, PAM
+  and finally the user.
+
+Data structures
+***************
+This section will define the data needed to obtain and verify the assertion.
+
+Assertion request data
+++++++++++++++++++++++
+To obtain the assertion the following information is required:
+
+* Domain (String)
+
+* List of credential IDs in b64 (String)
+
+* User-verification (Int)
+
+* Cryptographic challenge (String)
+
+.. _assertion-data:
+
+Assertion data
+++++++++++++++
+To verify the assertion the following informations is required:
+
+* Username (String)
+
+* Domain (String)
+
+* Used credential ID in b64 (String)
+
+* Public key in b64 (String)
+
+* COSE type (String)
+
+* User-verification (Int)
+
+* Cryptographic challenge (String)
+
+* Authenticator data in b64 (String)
+
+* Assertion signature in b64 (String)
+
+.. _kerberos-kdc-ipa-otpd:
+
+Kerberos KDC - ipa-otpd
++++++++++++++++++++++++
+The Kerberos KDC is expected to provide the following json-formatted string
```

Comment:
The messages are different, see description in https://github.com/SSSD/sssd/pull/6462

Reply:
Updated